src/cluster/src/check/mod.rs

```diff
@@ -379,16 +398,208 @@ impl ClusterChecker {
             Box::new(CreateServiceAccountPermission),
             Box::new(LoadBalancer),
         ];
+        self.checks.extend(checks);
+        self
+    }
 
-        // Collect results from running checks
-        let mut results = vec![];
+    /// Adds all checks required for starting a cluster on minikube.
+    ///
+    /// Note that no checks are run until [`run`] or [`run_with_progress`].
+    pub fn with_minikube_checks(mut self) -> Self {
+        let checks: Vec<Box<(dyn ClusterCheck)>> = vec![
+            Box::new(LoadableConfig),
+            Box::new(HelmVersion),
+            Box::new(SysChart),
+            Box::new(AlreadyInstalled),
+            Box::new(LoadBalancer),
+        ];
+        self.checks.extend(checks);
+        self
+    }
+
+    /// Adds all checks required for starting a local cluster.
+    ///
+    /// Note that no checks are run until [`run`] or [`run_with_progress`].
+    pub fn with_local_checks(mut self) -> Self {
+        let checks: Vec<Box<(dyn ClusterCheck)>> = vec![
+            Box::new(HelmVersion),
+            Box::new(K8Version),
+            Box::new(LoadableConfig),
+            Box::new(SysChart),
+        ];
+        self.checks.extend(checks);
+        self
+    }
 
-        for check in checks {
+    /// Performs all checks sequentially and returns the results when done.
+    ///
+    /// This may appear to "hang" if there are many checks. In order to see
+    /// fine-grained progress about ongoing checks, use [`run_with_progress`]
+    /// instead.
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use fluvio_cluster::{ClusterChecker, CheckResults};
+    /// # async fn do_run() {
+    /// let check_results: CheckResults = ClusterChecker::empty()
+    ///     .with_preflight_checks()
+    ///     .run()
+    ///     .await;
+    /// # }
+    /// ```
+    pub async fn run(&self) -> CheckResults {
```

Comment:
maybe changed this to `run_and_wait` or `run_and_collect` to make it clear this is synchronous or batch processing